.. index:: pair: class; cv::Exception
.. _doxid-d7/d6f/classcv_1_1_exception:

class cv::Exception
===================

.. toctree::
	:hidden:

Overview
~~~~~~~~

Class passed to an error. :ref:`More...<details-d7/d6f/classcv_1_1_exception>`


.. ref-code-block:: cpp
	:class: doxyrest-overview-code-block

	#include <core.hpp>
	
	class Exception: public std::exception
	{
	public:
		// fields
	
		int :ref:`code<doxid-d7/d6f/classcv_1_1_exception_1a81405b438c930b556e9c71210d799e2b>`;
		:ref:`String<doxid-dc/d0e/classcv_1_1_string>` :ref:`err<doxid-d7/d6f/classcv_1_1_exception_1a9d60908aaaad08b135bc3d8cea4dd676>`;
		:ref:`String<doxid-dc/d0e/classcv_1_1_string>` :ref:`file<doxid-d7/d6f/classcv_1_1_exception_1afef67d957832b1e20e24208a1da57379>`;
		:ref:`String<doxid-dc/d0e/classcv_1_1_string>` :ref:`func<doxid-d7/d6f/classcv_1_1_exception_1a5436ea62554761441ed4ce03938b49c1>`;
		int :ref:`line<doxid-d7/d6f/classcv_1_1_exception_1aa81481ded6391d046768d9829c56fd90>`;
		:ref:`String<doxid-dc/d0e/classcv_1_1_string>` :ref:`msg<doxid-d7/d6f/classcv_1_1_exception_1a7ead458b4fc88ebca20b231383c49845>`;

		// construction
	
		:ref:`Exception<doxid-d7/d6f/classcv_1_1_exception_1a59759ca13add62a2bdd1fb7605d675dc>`();
	
		:ref:`Exception<doxid-d7/d6f/classcv_1_1_exception_1a95c2117604b1d51c17e5ccd540f448b9>`(
			int _code,
			const :ref:`String<doxid-dc/d0e/classcv_1_1_string>`& _err,
			const :ref:`String<doxid-dc/d0e/classcv_1_1_string>`& _func,
			const :ref:`String<doxid-dc/d0e/classcv_1_1_string>`& _file,
			int _line
			);

		// methods
	
		void
		:target:`formatMessage<doxid-d7/d6f/classcv_1_1_exception_1afe6c2198ca3815de5385fd93f81f2c34>`();
	
		virtual
		const char*
		:ref:`what<doxid-d7/d6f/classcv_1_1_exception_1ab8d31c05241bb4fdc9dbd135fb8951fd>`() const;
	};

	// direct descendants

	class :ref:`FLANNException<doxid-d2/d20/classcvflann_1_1_f_l_a_n_n_exception>`;
.. _details-d7/d6f/classcv_1_1_exception:

Detailed Documentation
~~~~~~~~~~~~~~~~~~~~~~

Class passed to an error.

This class encapsulates all or almost all necessary information about the error happened in the program. The exception is usually constructed and thrown implicitly via CV_Error and CV_Error\_ macros.



.. rubric:: See also:

:ref:`error <doxid-db/de0/group__core__utils_1gacbd081fdb20423a63cf731569ba70b2b>`

Fields
------

.. index:: pair: variable; code
.. _doxid-d7/d6f/classcv_1_1_exception_1a81405b438c930b556e9c71210d799e2b:

.. ref-code-block:: cpp
	:class: doxyrest-title-code-block

	int code

error code



.. rubric:: See also:

:ref:`CVStatus <doxid-d2/df8/group__core__c_1gaeb7540290ce627c4b4e94203816e124c>`

.. index:: pair: variable; err
.. _doxid-d7/d6f/classcv_1_1_exception_1a9d60908aaaad08b135bc3d8cea4dd676:

.. ref-code-block:: cpp
	:class: doxyrest-title-code-block

	:ref:`String<doxid-dc/d0e/classcv_1_1_string>` err

error description

.. index:: pair: variable; file
.. _doxid-d7/d6f/classcv_1_1_exception_1afef67d957832b1e20e24208a1da57379:

.. ref-code-block:: cpp
	:class: doxyrest-title-code-block

	:ref:`String<doxid-dc/d0e/classcv_1_1_string>` file

source file name where the error has occurred

.. index:: pair: variable; func
.. _doxid-d7/d6f/classcv_1_1_exception_1a5436ea62554761441ed4ce03938b49c1:

.. ref-code-block:: cpp
	:class: doxyrest-title-code-block

	:ref:`String<doxid-dc/d0e/classcv_1_1_string>` func

function name. Available only when the compiler supports getting it

.. index:: pair: variable; line
.. _doxid-d7/d6f/classcv_1_1_exception_1aa81481ded6391d046768d9829c56fd90:

.. ref-code-block:: cpp
	:class: doxyrest-title-code-block

	int line

line number in the source file where the error has occurred

.. index:: pair: variable; msg
.. _doxid-d7/d6f/classcv_1_1_exception_1a7ead458b4fc88ebca20b231383c49845:

.. ref-code-block:: cpp
	:class: doxyrest-title-code-block

	:ref:`String<doxid-dc/d0e/classcv_1_1_string>` msg

the formatted error message

Construction
------------

.. index:: pair: function; Exception
.. _doxid-d7/d6f/classcv_1_1_exception_1a59759ca13add62a2bdd1fb7605d675dc:

.. ref-code-block:: cpp
	:class: doxyrest-title-code-block

	Exception()

Default constructor

.. index:: pair: function; Exception
.. _doxid-d7/d6f/classcv_1_1_exception_1a95c2117604b1d51c17e5ccd540f448b9:

.. ref-code-block:: cpp
	:class: doxyrest-title-code-block

	Exception(
		int _code,
		const :ref:`String<doxid-dc/d0e/classcv_1_1_string>`& _err,
		const :ref:`String<doxid-dc/d0e/classcv_1_1_string>`& _func,
		const :ref:`String<doxid-dc/d0e/classcv_1_1_string>`& _file,
		int _line
		)

Full constructor. Normally the constructor is not called explicitly. Instead, the macros :ref:`CV_Error() <doxid-db/de0/group__core__utils_1ga5b48c333c777666e076bd7052799f891>`, :ref:`CV_Error\_() <doxid-db/de0/group__core__utils_1ga1c0cd6e5bd9a5f915c6cab9c0632f969>` and :ref:`CV_Assert() <doxid-db/de0/group__core__utils_1gaf62bcd90f70e275191ab95136d85906b>` are used.

Methods
-------

.. index:: pair: function; what
.. _doxid-d7/d6f/classcv_1_1_exception_1ab8d31c05241bb4fdc9dbd135fb8951fd:

.. ref-code-block:: cpp
	:class: doxyrest-title-code-block

	virtual
	const char*
	what() const



.. rubric:: Returns:

the error description and the context as a text string.

